src/sdam/srv_polling/mod.rs

```diff
@@ -58,23 +54,34 @@ impl SrvPollingMonitor {
         });
     }
 
+    fn rescan_interval(&self) -> Duration {
+        std::cmp::max(self.rescan_interval, MIN_RESCAN_SRV_INTERVAL)
+    }
+
     async fn execute(&mut self) {
+        fn should_poll(tt: TopologyType) -> bool {
+            matches!(tt, TopologyType::Sharded | TopologyType::Unknown)
+        }
+
         while self.topology.is_alive() {
+            RUNTIME.delay_for(self.rescan_interval()).await;
```

Comment:
The issue here was that we were not waiting at all after the client was created to start SRV polling. This is an issue because the very first SRV lookup happens before client creation in `ClientOptions::parse`. So the order of events could end up like this:
1. `ClientOptions::parse("mongodb+srv://...").await`
2. `Client::with_options(options)`
3. SDAM monitor for `<address resolved from SRV records>` starts
4. SRV polling starts
5. Monitor check succeeds, updates topology with `localhost:27017`, `localhost:27018`, and `localhost:27019` and topology type ReplicaSetWithPrimary, removes `<address resolved from SRV records>` from topology
6. SRV poll finishes, replaces all hosts with `<address resolved from SRV records>` again
7. Repeat

By adding this wait to the beginning, we allow the SDAM monitors to discover that the topology is a replica set, and if so the SRV polling monitor will not attempt to update the topology.